web/html/src/manager/state/highstate.js

```diff
@@ -0,0 +1,198 @@
+            earliest: Functions.Utils.dateWithTimezone(localTime),
+            test: false
+        };
+        this.state = state;
+    }
+
+    applyHighstate = () => {
+        const request = Network.post(
+            "/rhn/manager/api/states/applyall",
+            JSON.stringify({
+                ids: minions.map(m => m.id),
+                earliest: Formats.LocalDateTime(this.state.earliest),
+                actionChain: this.state.actionChain ? this.state.actionChain.text : null,
+                test: this.state.test
+            }),
+            "application/json"
+        ).promise.then(data => {
+            const msg = MessagesUtils.info(this.state.actionChain ?
+                    <span>{t("Action has been successfully added to the ")}<ActionChainLink id={data}>{this.state.actionChain ? this.state.actionChain.text : ""}</ActionChainLink></span> :
+                    <span>{t("Applying the highstate has been ")}<ActionLink id={data}>{t("scheduled.")}</ActionLink></span>);
+
+            const msgs = this.state.messages.concat(msg);
+
+            // Do not spam UI showing old messages
+            while (msgs.length > messagesCounterLimit) {
+              msgs.shift();
+            }
+
+            this.setState({
+                messages: msgs
+            });
+        }).catch(this.handleResponseError);
+
+        return request;
+    };
+
+    handleResponseError = (jqXHR) => {
+      this.setState({
+           messages: Network.responseErrorMessage(jqXHR)
+      });
+    };
+
+    onDateTimeChanged = (date) => {
+        this.setState({"earliest": date});
+    };
+
+    onActionChainChanged = (actionChain) => {
+        this.setState({actionChain: actionChain})
+    };
+
+    onMessageChanged = (message) => {
+        this.setState({messages: message});
+    };
+
+    handleForwardAction = (action) => {
+        const loc = window.location;
+        if(action === "back") {
+            history.pushState(null, null, loc.pathname + loc.search);
+            this.setState({
+               action: undefined
+            });
+        } else {
+            const pathname = loc.pathname.replace("highstate", "states/schedules");
+            Functions.Utils.urlBounce(pathname + loc.search);
+        }
+    };
+
+    toggleTestState = () => {
+        this.setState({test: !this.state.test})
+    };
+
+    isSSM = () => {
+        return !!window.location.pathname.match("/ssm/");
+    };
+
+    updateSchedule =(schedule) => {
+        return Network.post(
+            "/rhn/manager/api/recurringactions/save",
+            JSON.stringify(schedule),
+            "application/json"
+        ).promise.then((data) => {
+            // HACK: propagate the errors from messages to the UI
+            let newMsgs = [];
+            if (data.messages === undefined || data.messages.length === 0) { // no errors from the server
+                newMsgs = MessagesUtils.info(<span>{t("Schedule successully created.")}</span>);
```

Comment:
So we don't receive any feedback from the server if everything went well? I'd prefer to get even just a `boolean` but something that cannot be misunderstood, even just a `{ success: true }` and we check this flag. If then `success` is `false` we look for `messages`

Reply:
Fixed.